### ISSUE TITLE
onnx_tf creates a Dilation2d layer which is not supported by TensorFlow Lite

### ISSUE THREAD

**Issue by devidlatkin** [onnx/onnx-tensorflow#595]
I use a docker image from here https://github.com/onnx/onnx-tensorflow/blob/master/README.md to transform my image from PyTorch to ONNX first, then to Frozen Graph and to tflite after. 
After transforming PyTorch to ONNX I have the next model summary acquired from print(onnx.helper.printable_graph(model_onnx.graph)) :

graph torch-jit-export (
  %test_input[FLOAT, 1x1x240x320]
) initializers (
  %conv1a.bias[FLOAT, 64]
  %conv1a.weight[FLOAT, 64x1x3x3]
  %conv1b.bias[FLOAT, 64]
  %conv1b.weight[FLOAT, 64x64x3x3]
  %conv2a.bias[FLOAT, 64]
  %conv2a.weight[FLOAT, 64x64x3x3]
  %conv2b.bias[FLOAT, 64]
  %conv2b.weight[FLOAT, 64x64x3x3]
  %conv3a.bias[FLOAT, 128]
  %conv3a.weight[FLOAT, 128x64x3x3]
  %conv3b.bias[FLOAT, 128]
  %conv3b.weight[FLOAT, 128x128x3x3]
  %conv4a.bias[FLOAT, 128]
  %conv4a.weight[FLOAT, 128x128x3x3]
  %conv4b.bias[FLOAT, 128]
  %conv4b.weight[FLOAT, 128x128x3x3]
  %convDa.bias[FLOAT, 256]
  %convDa.weight[FLOAT, 256x128x3x3]
  %convDb.bias[FLOAT, 256]
  %convDb.weight[FLOAT, 256x256x1x1]
  %convPa.bias[FLOAT, 256]
  %convPa.weight[FLOAT, 256x128x3x3]
  %convPb.bias[FLOAT, 65]
  %convPb.weight[FLOAT, 65x256x1x1]
) {
  %25 = Conv[dilations = [1, 1], group = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [1, 1]](%input, %conv1a.weight, %conv1a.bias)
  %26 = Relu(%25)
  %27 = Conv[dilations = [1, 1], group = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [1, 1]](%26, %conv1b.weight, %conv1b.bias)
  %28 = Relu(%27)
  %29 = MaxPool[kernel_shape = [2, 2], pads = [0, 0, 0, 0], strides = [2, 2]](%28)
  %30 = Conv[dilations = [1, 1], group = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [1, 1]](%29, %conv2a.weight, %conv2a.bias)
  %31 = Relu(%30)
  %32 = Conv[dilations = [1, 1], group = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [1, 1]](%31, %conv2b.weight, %conv2b.bias)
  %33 = Relu(%32)
  %34 = MaxPool[kernel_shape = [2, 2], pads = [0, 0, 0, 0], strides = [2, 2]](%33)
  %35 = Conv[dilations = [1, 1], group = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [1, 1]](%34, %conv3a.weight, %conv3a.bias)
  %36 = Relu(%35)
  %37 = Conv[dilations = [1, 1], group = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [1, 1]](%36, %conv3b.weight, %conv3b.bias)
  %38 = Relu(%37)
  %39 = MaxPool[kernel_shape = [2, 2], pads = [0, 0, 0, 0], strides = [2, 2]](%38)
  %40 = Conv[dilations = [1, 1], group = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [1, 1]](%39, %conv4a.weight, %conv4a.bias)
  %41 = Relu(%40)
  %42 = Conv[dilations = [1, 1], group = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [1, 1]](%41, %conv4b.weight, %conv4b.bias)
  %43 = Relu(%42)
  %44 = Conv[dilations = [1, 1], group = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [1, 1]](%43, %convPa.weight, %convPa.bias)
  %45 = Relu(%44)
  %test_output_kps = Conv[dilations = [1, 1], group = 1, kernel_shape = [1, 1], pads = [0, 0, 0, 0], strides = [1, 1]](%45, %convPb.weight, %convPb.bias)
  %47 = Conv[dilations = [1, 1], group = 1, kernel_shape = [3, 3], pads = [1, 1, 1, 1], strides = [1, 1]](%43, %convDa.weight, %convDa.bias)
  %48 = Relu(%47)
  %49 = Conv[dilations = [1, 1], group = 1, kernel_shape = [1, 1], pads = [0, 0, 0, 0], strides = [1, 1]](%48, %convDb.weight, %convDb.bias)
  %50 = ReduceL2[axes = [1], keepdims = 0](%49)
  %51 = Unsqueeze[axes = [1]](%50)
  %test_output_desc = Div(%49, %51)
  return %output1, %output2
}

After transforming my ONNX model into Frozen Graph pb file I have the next output acquired from 
for op in frozen_graph_def.get_operations():
    print(op.name)

prefix/conv1a.bias
prefix/conv1a.weight
prefix/conv1b.bias
prefix/conv1b.weight
prefix/conv2a.bias
prefix/conv2a.weight
prefix/conv2b.bias
prefix/conv2b.weight
prefix/conv3a.bias
prefix/conv3a.weight
prefix/conv3b.bias
prefix/conv3b.weight
prefix/conv4a.bias
prefix/conv4a.weight
prefix/conv4b.bias
prefix/conv4b.weight
prefix/convDa.bias
prefix/convDa.weight
prefix/convDb.bias
prefix/convDb.weight
prefix/convPa.bias
prefix/convPa.weight
prefix/convPb.bias
prefix/convPb.weight
prefix/input
prefix/transpose/perm
prefix/transpose
prefix/Const
prefix/Pad
prefix/Const_1
prefix/split/split_dim
prefix/split
prefix/transpose_1/perm
prefix/transpose_1
prefix/Const_2
prefix/split_1/split_dim
prefix/split_1
prefix/convolution
prefix/concat/concat_dim
prefix/concat/concat
prefix/Add
prefix/transpose_2/perm
prefix/transpose_2
prefix/Relu
prefix/transpose_3/perm
prefix/transpose_3
prefix/Const_3
prefix/Pad_1
prefix/Const_4
prefix/split_2/split_dim
prefix/split_2
prefix/transpose_4/perm
prefix/transpose_4
prefix/Const_5
prefix/split_3/split_dim
prefix/split_3
prefix/convolution_1
prefix/concat_1/concat_dim
prefix/concat_1/concat
prefix/Add_1
prefix/transpose_5/perm
prefix/transpose_5
prefix/Relu_1
prefix/transpose_6/perm
prefix/transpose_6
prefix/zeros
prefix/Dilation2D
prefix/transpose_7/perm
prefix/transpose_7
prefix/transpose_8/perm
prefix/transpose_8
prefix/Const_6
prefix/Pad_2
prefix/Const_7
prefix/split_4/split_dim
prefix/split_4
prefix/transpose_9/perm
prefix/transpose_9
prefix/Const_8
prefix/split_5/split_dim
prefix/split_5
prefix/convolution_2
prefix/concat_2/concat_dim
prefix/concat_2/concat
prefix/Add_2
prefix/transpose_10/perm
prefix/transpose_10
prefix/Relu_2
prefix/transpose_11/perm
prefix/transpose_11
prefix/Const_9
prefix/Pad_3
prefix/Const_10
prefix/split_6/split_dim
prefix/split_6
prefix/transpose_12/perm
prefix/transpose_12
prefix/Const_11
prefix/split_7/split_dim
prefix/split_7
prefix/convolution_3
prefix/concat_3/concat_dim
prefix/concat_3/concat
prefix/Add_3
prefix/transpose_13/perm
prefix/transpose_13
prefix/Relu_3
prefix/transpose_14/perm
prefix/transpose_14
prefix/zeros_1
prefix/Dilation2D_1
prefix/transpose_15/perm
prefix/transpose_15
prefix/transpose_16/perm
prefix/transpose_16
prefix/Const_12
prefix/Pad_4
prefix/Const_13
prefix/split_8/split_dim
prefix/split_8
prefix/transpose_17/perm
prefix/transpose_17
prefix/Const_14
prefix/split_9/split_dim
prefix/split_9
prefix/convolution_4
prefix/concat_4/concat_dim
prefix/concat_4/concat
prefix/Add_4
prefix/transpose_18/perm
prefix/transpose_18
prefix/Relu_4
prefix/transpose_19/perm
prefix/transpose_19
prefix/Const_15
prefix/Pad_5
prefix/Const_16
prefix/split_10/split_dim
prefix/split_10
prefix/transpose_20/perm
prefix/transpose_20
prefix/Const_17
prefix/split_11/split_dim
prefix/split_11
prefix/convolution_5
prefix/concat_5/concat_dim
prefix/concat_5/concat
prefix/Add_5
prefix/transpose_21/perm
prefix/transpose_21
prefix/Relu_5
prefix/transpose_22/perm
prefix/transpose_22
prefix/zeros_2
prefix/Dilation2D_2
prefix/transpose_23/perm
prefix/transpose_23
prefix/transpose_24/perm
prefix/transpose_24
prefix/Const_18
prefix/Pad_6
prefix/Const_19
prefix/split_12/split_dim
prefix/split_12
prefix/transpose_25/perm
prefix/transpose_25
prefix/Const_20
prefix/split_13/split_dim
prefix/split_13
prefix/convolution_6
prefix/concat_6/concat_dim
prefix/concat_6/concat
prefix/Add_6
prefix/transpose_26/perm
prefix/transpose_26
prefix/Relu_6
prefix/transpose_27/perm
prefix/transpose_27
prefix/Const_21
prefix/Pad_7
prefix/Const_22
prefix/split_14/split_dim
prefix/split_14
prefix/transpose_28/perm
prefix/transpose_28
prefix/Const_23
prefix/split_15/split_dim
prefix/split_15
prefix/convolution_7
prefix/concat_7/concat_dim
prefix/concat_7/concat
prefix/Add_7
prefix/transpose_29/perm
prefix/transpose_29
prefix/Relu_7
prefix/transpose_30/perm
prefix/transpose_30
prefix/Const_24
prefix/Pad_8
prefix/Const_25
prefix/split_16/split_dim
prefix/split_16
prefix/transpose_31/perm
prefix/transpose_31
prefix/Const_26
prefix/split_17/split_dim
prefix/split_17
prefix/convolution_8
prefix/concat_8/concat_dim
prefix/concat_8/concat
prefix/Add_8
prefix/transpose_32/perm
prefix/transpose_32
prefix/Relu_8
prefix/transpose_33/perm
prefix/transpose_33
prefix/Const_27
prefix/split_18/split_dim
prefix/split_18
prefix/transpose_34/perm
prefix/transpose_34
prefix/Const_28
prefix/split_19/split_dim
prefix/split_19
prefix/convolution_9
prefix/concat_9/concat_dim
prefix/concat_9/concat
prefix/Add_9
prefix/transpose_35/perm
prefix/output1
prefix/transpose_36/perm
prefix/transpose_36
prefix/Const_29
prefix/Pad_9
prefix/Const_30
prefix/split_20/split_dim
prefix/split_20
prefix/transpose_37/perm
prefix/transpose_37
prefix/Const_31
prefix/split_21/split_dim
prefix/split_21
prefix/convolution_10
prefix/concat_10/concat_dim
prefix/concat_10/concat
prefix/Add_10
prefix/transpose_38/perm
prefix/transpose_38
prefix/Relu_9
prefix/transpose_39/perm
prefix/transpose_39
prefix/Const_32
prefix/split_22/split_dim
prefix/split_22
prefix/transpose_40/perm
prefix/transpose_40
prefix/Const_33
prefix/split_23/split_dim
prefix/split_23
prefix/convolution_11
prefix/concat_11/concat_dim
prefix/concat_11/concat
prefix/Add_11
prefix/transpose_41/perm
prefix/transpose_41
prefix/norm/mul
prefix/norm/Sum/reduction_indices
prefix/norm/Sum
prefix/norm/Sqrt
prefix/norm/Squeeze
prefix/ExpandDims/dim
prefix/ExpandDims
prefix/output2

**As you can see, onnx_tensorflow added a lot of additional layers/operations while transforming model from onnx. 
A particular layer which causes some troubles for me is Dilation2D** 
**This layer is not supported by TensorFlow Lite, thus I simply can't normally transform my pb model to tflite.**

Do you have any idea why onnx_tensorflow use Dilation2D despite the fact, that I don't have it in my original architecture?

Thank you

**Comment by theotheo** [comment]
I have the same issue. Is there a workaround?

**Comment by Wheest** [comment]
Also have a similar issue, with this operation, when trying to export to TF Lite.  All of my dilations are [1,1], so should not need to be included as an operation.

**Comment by sdmonov** [comment]
@devidlatkin @theotheo @Wheest can you test #613. it should solve the issue with Dilations2D used instead of MaxPool.

**Comment by skavulya** [comment]
@sdmonov  I experienced the same issue with Dilations2D and I have successfully converted my onnx model to tflite with this fix. Thanks for the fix!

**Comment by devidlatkin** [comment]
@sdmonov Thank you for response! 
I solved my issed by converting weights manually, but I will try your approach.

Thank you!
